## Use BioThings_client to get info from drugbank

In [1]:
import biothings_client

In [2]:
# use the mychem client
mc = biothings_client.get_client('drug')

In [6]:
# get all clinically approved small molecule drugs in drugbank
# chembl.molecular_type:"small molecule' will get all drugs which are categorized as small molecule
# drugbank.groups:approved will get all drugs which are approved
# fields='puchem.cid' will get the pubchem ID for these drugs
# fetch_all=True will get all drugs, if not specified, the client will only return first 10 hits by default
data = mc.query('chembl.molecule_type:"Small molecule" AND drugbank.groups:approved', fields='pubchem.cid', fetch_all=True)

In [7]:
# the results from python client will be returned as a python generator
data

<generator object _fetch_all at 0x10d317d00>

In [8]:
results = list(data)

Fetching 1559 drug(s) . . .
No results to return


**Results**: Total number of clinically approved small molecule drugs are 1559

In [9]:
results[1]

{'_id': 'SNICXCGAKADSCV-UHFFFAOYSA-N',
 '_score': 12.787065,
 'pubchem': {'cid': 'CID942'}}

In [11]:
pubchem_list = []
for _doc in results:
    if 'pubchem' in _doc and 'cid' in _doc['pubchem']:
        pubchem_list.append(_doc['pubchem']['cid'])

In [12]:
len(pubchem_list)

1517

**Results**: of the 1559 drugs found in MyChem.info, 1517 of them have pubchem IDs.